src/EnergyPlus/DataHeatBalance.hh

```diff
@@ -1329,39 +1331,39 @@ namespace DataHeatBalance {
-        Real64 QSECON = 0.0;  // ENERGY CONVECTED FROM Steam EQUIPMENT
-        Real64 QSERAD = 0.0;  // ENERGY RADIATED FROM Steam EQUIPMENT
-        Real64 QSELost = 0.0; // Energy from Steam Equipment (lost)
-        Real64 QSELAT = 0.0;  // LATENT ENERGY FROM Steam Equipment
-        Real64 QBBCON = 0.0;  // ENERGY CONVECTED FROM BASEBOARD HEATING
-        Real64 QBBRAD = 0.0;  // ENERGY RADIATED FROM BASEBOARD HEATING
+        Real64 NOFOCC = 0.0;                                                            // Number of Occupants
+        Real64 QOCTOT = 0.0;                                                            // Total Energy from Occupants
+        Real64 QOCSEN = 0.0;                                                            // Sensible Energy from Occupants
+        Real64 QOCCON = 0.0;                                                            // ENERGY CONVECTED FROM OCCUPANTS (WH)
+        Real64 QOCRAD = 0.0;                                                            // ENERGY RADIATED FROM OCCUPANTS
+        Real64 QOCLAT = 0.0;                                                            // LATENT ENERGY FROM OCCUPANTS
+        Real64 QLTTOT = 0.0;                                                            // TOTAL ENERGY INTO LIGHTS (WH)
+        Real64 QLTCON = 0.0;                                                            // ENERGY CONVECTED TO SPACE AIR FROM LIGHTS
+        Real64 QLTRAD = 0.0;                                                            // ENERGY RADIATED TO SPACE FROM LIGHTS
+        Real64 QLTCRA = 0.0;                                                            // ENERGY CONVECTED TO RETURN AIR FROM LIGHTS
+        Real64 QLTSW = 0.0;                                                             // VISIBLE ENERGY FROM LIGHTS
+        Real64 QEECON = 0.0;                                                            // ENERGY CONVECTED FROM ELECTRIC EQUIPMENT
+        Real64 QEERAD = 0.0;                                                            // ENERGY RADIATED FROM ELECTRIC EQUIPMENT
+        Real64 QEELost = 0.0;                                                           // Energy from Electric Equipment (lost)
+        Real64 QEELAT = 0.0;                                                            // LATENT ENERGY FROM Electric Equipment
+        Real64 QGECON = 0.0;                                                            // ENERGY CONVECTED FROM GAS EQUIPMENT
+        Real64 QGERAD = 0.0;                                                            // ENERGY RADIATED FROM GAS EQUIPMENT
+        Real64 QGELost = 0.0;                                                           // Energy from Gas Equipment (lost)
+        Real64 QGELAT = 0.0;                                                            // LATENT ENERGY FROM Gas Equipment
+        std::array<Real64, (int)ExteriorEnergyUse::ExteriorFuelUsage::Num + 1> QOECON;  // ENERGY CONVECTED FROM OTHER EQUIPMENT
```

Comment:
Why make these arrays? These are not currently fuel-type-specific, and I don't see new fuel-type output variables for these values here.

Reply:
In order to report OtherPower and OtherConsump by fuel type, their individual components (QOECON, etc.) also need to be  stored by fuel type, no? See [this calculation](https://github.com/NREL/EnergyPlus/blob/04cfff16fed7c144838c1cae5b01b6271a6b906c/src/EnergyPlus/InternalHeatGains.cc#L8444-L8447) for example. I don't see how you could do the calculations if QOECON, etc. are not arrays.

Reply:
Yuck. That seems like extra work. We know the other eq power by type and should be able to use that rather than re-summing the gain components. If you don't mind, may I take a crack at refactoring the zone and space report loops here? I think this section could be more efficient.

Reply:
Please feel free to have at it! You are definitely more familiar with the code.